### PR TITLE
[ts-sdk] Support null and undefined as absense of value in option

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
The previous implementation supported `undefined`.  It occurred that it may want to also support `null` for an empty option.

### Test Plan
Unit test

### CODE REVIEW

File: ecosystem/typescript/sdk/src/transaction_builder/builder_utils.ts [hunk 1]
```diff
@@ -161,7 +161,7 @@ function serializeStruct(argVal: any, argType: TypeTag, serializer: Serializer,
 function serializeOption(argVal: any, argType: TypeTag, serializer: Serializer, depth: number) {
   // For option, we determine if it's empty or not empty first
   // empty option is nothing, we specifically check for undefined to prevent fuzzy matching
-  if (argVal === undefined) {
+  if (argVal === undefined || argVal === null) {
```
Comment (0xmaayan) [inline ecosystem/typescript/sdk/src/transaction_builder/builder_utils.ts:164]: might be easier
```suggestion
  if (!argVal) {
```

**Review by 0xmaayan — CHANGES_REQUESTED** [review]
please remove yarn-error.log and package-lock.json files

File: ecosystem/typescript/sdk/src/transaction_builder/builder_utils.ts (hunk 1 above)
Comment (gregnazario) [reply ecosystem/typescript/sdk/src/transaction_builder/builder_utils.ts:164]: We have be very careful about things like this, and probably should audit them.

This argval could be `""` or `0`, which are both falsey.

**Review by 0xmaayan — APPROVED** [review]

File: ecosystem/typescript/sdk/src/transaction_builder/builder_utils.ts (hunk 1 above)
Comment (hardsetting) [reply ecosystem/typescript/sdk/src/transaction_builder/builder_utils.ts:164]: +1 for being explicit

**Review by hardsetting — APPROVED** [review]
LGTM

**Pull request merged**